File: MyApplication2/app/src/main/java/com/mchi/myapplication/MainActivity.java
package com.example.registronotas;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText etNombre, etNota1, etNota2, etNota3;
    Button btnCalcular;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etNombre = findViewById(R.id.etNombre);
        etNota1 = findViewById(R.id.etNota1);
        etNota2 = findViewById(R.id.etNota2);
        etNota3 = findViewById(R.id.etNota3);
        btnCalcular = findViewById(R.id.btnCalcular);

        btnCalcular.setOnClickListener(v -> {
            String nombre = etNombre.getText().toString().trim();
            String s1 = etNota1.getText().toString().trim();
            String s2 = etNota2.getText().toString().trim();
            String s3 = etNota3.getText().toString().trim();

            if (nombre.isEmpty() || s1.isEmpty() || s2.isEmpty() || s3.isEmpty()) {
                Toast.makeText(this, "Complete todos los campos", Toast.LENGTH_SHORT).show();
                return;
            }

            float n1 = Float.parseFloat(s1);
            float n2 = Float.parseFloat(s2);
            float n3 = Float.parseFloat(s3);

            if (n1 < 0 || n1 > 20 || n2 < 0 || n2 > 20 || n3 < 0 || n3 > 20) {
                Toast.makeText(this, "Las notas deben estar entre 0 y 20", Toast.LENGTH_SHORT).show();
                return;
            }

            float promedio = (n1 + n2 + n3) / 3;

            Intent intent = new Intent(this, ResultadoActivity.class);
            intent.putExtra("NOMBRE", nombre);
            intent.putExtra("PROMEDIO", promedio);
            startActivity(intent);
        });
    }
}

File: Ej1/app/src/main/java/com/mchi/conversorvario/MainActivity.kt
package com.mchi.conversorvario

import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.Spinner
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var editTextCantidad: EditText
    private lateinit var spinner1: Spinner
    private lateinit var spinner2: Spinner
    private val monedas: List<Moneda> = listOf(Dolar(), Sol(), Euro(), Libra(), Real(), Peso(), Rupia(), Yen(), Yuan())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // Inicializar vistas
        editTextCantidad = findViewById(R.id.editTextNumberDecimal)
        spinner1 = findViewById(R.id.spinner1)
        spinner2 = findViewById(R.id.spinner2)
        val nombres = monedas.map {it.nombre}
        val adapter = ArrayAdapter(this, R.layout.spinners, nombres)
        adapter.setDropDownViewResource(R.layout.spinners)
        spinner1.adapter = adapter
        spinner2.adapter = adapter
    }

    private fun convertirMoneda() {
        val cantidadTexto = editTextCantidad.text.toString().trim()

        // Validación mejorada
        if (cantidadTexto.isEmpty()) {
            mostrarError("Ingresa una cantidad")
            return
        }
        val cantidad = cantidadTexto.toFloatOrNull()
        if (cantidad == null || cantidad <= 0) {
            mostrarError("Ingresa un número válido mayor a 0")
            return
        }

        val monedaO = monedas[spinner1.selectedItemPosition]
        val monedaD = monedas[spinner2.selectedItemPosition]
        if (monedaO.nombre == monedaD.nombre) {
            mostrarError("Selecciona monedas diferentes")
            return
        }
        val enDolares = cantidad / monedaD.tasaADolar
        val resultado = enDolares * monedaO.tasaADolar
        mostrarResultado("${"%.2f".format(cantidad)}${monedaO.simbolo} =${"%.2f".format(resultado)} s${monedaD.simbolo}")

    }
    private fun switchearMonedas(){
        val monedaOpos = spinner1.selectedItemPosition
        val monedaDpos = spinner2.selectedItemPosition
        if (monedaOpos == monedaDpos) {
            mostrarError("Selecciona monedas diferentes")
            return
        }
        spinner1.setSelection(monedaDpos)
        spinner2.setSelection(monedaOpos)
    }
    fun miClicManejador(view: View) {
        when (view.id) {
            R.id.btnConvertir -> convertirMoneda()
            R.id.btnAlternar -> switchearMonedas()
        }
    }

    private fun mostrarError(mensaje: String) {
        Toast.makeText(this, mensaje, Toast.LENGTH_LONG).show()
        editTextCantidad.requestFocus()
    }
    private fun mostrarResultado(mensaje: String) {
        Toast.makeText(this, mensaje, Toast.LENGTH_LONG).show()
    }
    }